src/commands/appSettings/filterDownloadAppSettings.ts

```diff
@@ -4,65 +4,61 @@
-    let suppressPrompt: boolean = false;
-    let overwriteSetting: boolean = false;
-    let overwriteSettingsToIgnore: boolean = false;
 
     const addedKeys: string[] = [];
     const updatedKeys: string[] = [];
     const userIgnoredKeys: string[] = [];
     const matchingKeys: string[] = [];
-    const securitySettingsIgnored: string[] = [];
-    const listOfSettingsToIgnore: string[] = ["AzureWebJobsStorage", "WEBSITE_CONTENTAZUREFILECONNECTIONSTRING", "WEBSITE_CONTENTSHARE"];
-
-    const yesToAll: vscode.MessageItem = { title: localize('yesToAll', 'Yes to all') };
-    const noToAll: vscode.MessageItem = { title: localize('noToAll', 'No to all') };
 
+    const listOfSettingsToIgnore: string[] = ["AzureWebJobsStorage", "WEBSITE_CONTENTAZUREFILECONNECTIONSTRING", "WEBSITE_CONTENTSHARE"];
+    const userChosenSettings: string[] = [];
+    const options: vscode.QuickPickItem[] = [];
     destinationSettingsToIgnore.length = 0;
 
-    for (const key of Object.keys(sourceSettings)) {
-        if (listOfSettingsToIgnore.includes(key)) {
-            if (!suppressPrompt) {
-                const message: string = localize('overwriteSettingsToIgnore', 'Setting "{0}" contains secrets. Download?', key);
-                const result: vscode.MessageItem = await context.ui.showWarningMessage(message, { modal: true }, DialogResponses.yes, yesToAll, DialogResponses.no, noToAll);
-                overwriteSettingsToIgnore = result === DialogResponses.yes || result === yesToAll;
-                suppressPrompt = result === yesToAll || result === noToAll;
-            }
-            if (overwriteSettingsToIgnore) {
-                updatedKeys.push(key);
-                destinationSettings[key] = sourceSettings[key];
-            } else {
-                securitySettingsIgnored.push(key);
-                destinationSettings[key] = "_REDACTED_";
-                destinationSettingsToIgnore.push(key);
-            }
+    for (const element of Object.keys(sourceSettings)) {
+        if (listOfSettingsToIgnore.includes(element)) {
+            options.push({
+                label: element
+            });
         }
         else {
-            if (destinationSettings[key] === undefined) { // Have an explicit check for undefined as empty settings should not pass the condition
+            options.push({
+                label: element,
+                picked: true
+            });
+        }
+    }
+
+    const result = await context.ui.showQuickPick(options, { placeHolder: 'Please select the app settings you would like to download:', canPickMany: true });
+
+    while (result.length != 0) {
+        const temp = result.pop();
+        if (temp === undefined) break;
+        userChosenSettings.push(temp?.label);
+    }
+
+    for (const key of Object.keys(sourceSettings)) {
+        if (userChosenSettings.includes(key)) {
+            if (destinationSettings[key] === undefined) { // when app setting does not exist locally
```

Comment:
in this case lets have the comment indicating that `destinationSettings[key]`  could be empty and that is valid.. thus you have a specific equality check for undefined.

Reply:
sure, how about "Explicit check for undefined as destinationSettings[key] could be empty but valid"?